frontend/src/metabase/metadata/components/DiscardFieldValuesButton/DiscardFieldValuesButton.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useRef, useState } from "react";
+import { t } from "ttag";
+
+import { useDiscardFieldValuesMutation } from "metabase/api";
+import { useDispatch } from "metabase/lib/redux";
+import { addUndo } from "metabase/redux/undo";
+import { Button } from "metabase/ui";
+import type { FieldId } from "metabase-types/api";
+
+interface Props {
+  fieldId: FieldId;
+}
+
+export const DiscardFieldValuesButton = ({ fieldId }: Props) => {
+  const dispatch = useDispatch();
+
+  const [started, setStarted] = useState(false);
+  const timeoutIdRef = useRef<number>();
+  const [discardFieldValues, { error }] = useDiscardFieldValuesMutation();
+
+  const handleClick = async () => {
+    const response = await discardFieldValues(fieldId);
+
+    if (!response.error) {
+      setStarted(true);
+
+      window.clearTimeout(timeoutIdRef.current);
+      timeoutIdRef.current = window.setTimeout(() => setStarted(false), 2000);
```

Comment:
This will cause a React error when the timeout triggers after the component unmounts.

Consider using a `useEffect` that clears the timeout when the component unmounts.

You might as well start the timeout there as well (although I get why you might want to keep it in the callback so multiple clicks can push back the timeout on each click):

```
useEffect(() => {
  if (!started) {
    return
  }
  const timeoutId = window.setTimeout(() => setStarted(false), 2000)
  return () => window.clearTimeout(timeoutId)
}, [started])
```

Reply:
dabfefba1383ec51d5f159442093c4961bb44f44